### PR TITLE
Hide voice-search icon if speech recognition is unavailable.

### PR DESCRIPTION
**Pull request by dbrant**
This converts the SearchCard in the feed to Kotlin, and makes a tiny fix:
- By request of Huawei: if voice recognition is not available on the device, then we should hide the Voice icon, which becomes useless.

### CODE REVIEW

File: app/src/main/java/org/wikipedia/feed/searchbar/SearchCardView.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import org.wikipedia.util.FeedbackUtil
+import org.wikipedia.util.ResourceUtil
+
+class SearchCardView(context: Context) : DefaultFeedCardView<SearchCard>(context) {
+    interface Callback {
+        fun onSearchRequested(view: View)
+        fun onVoiceSearchRequested()
+    }
+
+    private val binding = ViewSearchBarBinding.inflate(LayoutInflater.from(context), this, true)
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/feed/searchbar/SearchCardView.kt:19]: Can this be a local variable?

File: app/src/main/java/org/wikipedia/feed/searchbar/SearchCardView.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    init {
+        binding.searchContainer.setCardBackgroundColor(ResourceUtil.getThemedColor(context, R.attr.color_group_22))
+        FeedbackUtil.setButtonLongPressToast(findViewById(R.id.voice_search_button))
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/feed/searchbar/SearchCardView.kt:23]: Should we use `binding.voiceSearchButton` instead of `findViewById`?

**Review by cooltey — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/wikipedia/feed/searchbar/SearchCardView.kt (hunk 1 above)
Comment (dbrant) [reply app/src/main/java/org/wikipedia/feed/searchbar/SearchCardView.kt:19]: yep

File: app/src/main/java/org/wikipedia/feed/searchbar/SearchCardView.kt (hunk 2 above)
Comment (dbrant) [reply app/src/main/java/org/wikipedia/feed/searchbar/SearchCardView.kt:23]: whoops

**Pull request merged**